Source/Core/VideoBackends/Vulkan/StagingBuffer.cpp

```diff
@@ -180,59 +161,46 @@ bool StagingBuffer::AllocateBuffer(STAGING_BUFFER_TYPE type, VkDeviceSize size,
       0,                                     // uint32_t               queueFamilyIndexCount
       nullptr                                // const uint32_t*        pQueueFamilyIndices
   };
-  VkResult res =
-      vkCreateBuffer(g_vulkan_context->GetDevice(), &buffer_create_info, nullptr, out_buffer);
-  if (res != VK_SUCCESS)
-  {
-    LOG_VULKAN_ERROR(res, "vkCreateBuffer failed: ");
-    return false;
-  }
 
-  VkMemoryRequirements requirements;
-  vkGetBufferMemoryRequirements(g_vulkan_context->GetDevice(), *out_buffer, &requirements);
+  VmaAllocationCreateInfo alloc_create_info = {};
+  alloc_create_info.flags = 0;
```

Comment:
I believe this structure is used for readback.  Docs suggest this:

```
alloc_create_info.usage = VMA_MEMORY_USAGE_AUTO;
alloc_create_info.flags = VMA_ALLOCATION_CREATE_HOST_ACCESS_RANDOM_BIT |
    VMA_ALLOCATION_CREATE_MAPPED_BIT;
```

Reply:
I opted to provide VkMemoryPropertyFlags manually instead of using those flags. That way we can keep the workaround for slow incoherent memory on Mali GPUs. (see GetReadbackMemoryFlags)

Reply:
I do not know how the api decides between "flags", "required flags", and "preferred flags".  Other devs may disagree but I'd rather go with best practices than trying to cater our code to a particular vendor/driver.

That's not to say we can't support Mali but I'd rather use best practices and special case Mali.

For instance:

```
#ifdef WhateverPlatformMaliIsOn (android?)
  if (MaliWorkaround)
  {
    // Do something
  }
  else
  {
    // Do whatever we want to do
  }
#else
VmaAllocationCreateInfo alloc_create_info = {};
alloc_create_info.usage = VMA_MEMORY_USAGE_AUTO;
alloc_create_info.flags = VMA_ALLOCATION_CREATE_HOST_ACCESS_RANDOM_BIT | VMA_ALLOCATION_CREATE_MAPPED_BIT;
alloc_create_info.pool = VK_NULL_HANDLE;
alloc_create_info.pUserData = nullptr;
alloc_create_info.priority = 0.0;
#endif
```

(maybe break these into functions?)

We also avoid the branching cost on machines that don't need it.